src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeMetadataGenerator.cs

```diff
@@ -87,7 +93,14 @@ public byte[] GenerateEventMetadata(EventMetadata eventMetadata)
                 // Increase the metadataLength for parameters.
                 foreach (var parameter in parameters)
                 {
-                    metadataLength = metadataLength + parameter.GetMetadataLength();
+                    int pMetadataLength = parameter.GetMetadataLength();
+                    // The call above may return -1 which means we failed to get the metadata length. 
+                    // We then return a null metadata blob to prevent it from generating malformed metadata.
+                    if (pMetadataLength < 0)
```

Comment:
You can check for an invalid metadataLength here and if that happens just go with the default metadata.  You can also set ```parameters``` to an empty collection and that will allow you to not special case the generation of metadata below - you'll just get the default metadata with zero parameters.